src/vm.ts

```diff
@@ -29,25 +26,28 @@ export function createPluginConfigVM(
         },
         '__pluginHostMeta'
     )
-    vm.run(`const __pluginMeta = { ...__pluginHostMeta, ...__pluginLocalMeta };`)
-    vm.run(`${libJs} ; ${indexJs}`)
-    // remain backwards compatible with 1) compiled and non-compiled js, 2) setupPlugin and old setupTeam
-    vm.run(`;global.setupPlugin 
-            ? global.setupPlugin(__pluginMeta) 
-            : exports.setupPlugin 
-                ? exports.setupPlugin(__pluginMeta) 
-                : global.setupTeam 
-                    ? global.setupTeam(__pluginMeta) 
-                    : exports.setupTeam 
-                        ? exports.setupTeam(__pluginMeta) 
-                        : false;`)
+    vm.run(
+        `
+        const module = { exports: {} };
+        const exports = {};
+        const __pluginLocalMeta = { global: {} };
+        const __pluginMeta = { ...__pluginHostMeta, ...__pluginLocalMeta };
+        const __getGlobalWithMeta = (key) => {
+            const method = exports[key] || module.exports[key] || global[key];
+            if (!method) { return null };
+            return (...args) => method(...args, __pluginMeta)
+        } 
```

Comment:
It may be just me being thick, but I don't exactly grasp what all these "metas" do and how this works 🤔 Some comments here could be enlightening

Reply:
👍 . I tried making it a bit more clear.

Reply:
Thanks, that'd definitely clearer!